src/main/java/BMO.java

```diff
@@ -0,0 +1,321 @@
+            System.out.println(Constants.errorPrint.erroneousAdd());
+            receive();
+        }
+
+        throw new IllegalArgumentException("Unrecognized date and time format: " + input);
+    }
+
+    static void addLog(String input) {
+        // create string formats for each task type for format checking
+        String deadlineFormat = "^deadline\\s+(\\w+(\\s+\\w+)*)\\s+/by\\s+(\\S+(\\s+\\w+|/)*)$";
+        Pattern deadlinePattern = Pattern.compile(deadlineFormat);
+        Matcher deadlineMatcher = deadlinePattern.matcher(input);
+
+        String eventFormat = "^event\\s+(\\w+(\\s+\\w+)*)\\s+/from\\s+(\\S+(\\s+\\w+|/)*)\\s+/to\\s+(\\S+(\\s+\\w+|/)*)$";
+        Pattern eventPattern = Pattern.compile(eventFormat);
+        Matcher eventMatcher = eventPattern.matcher(input);
+
+        String toDoFormat = "^todo\\s+(\\S+(\\s+\\w+)*)$";
+        Pattern toDoPattern = Pattern.compile(toDoFormat);
+        Matcher toDoMatcher = toDoPattern.matcher(input);
+
+        Task newTask;
+        if (deadlineMatcher.matches()) {
+            String task = deadlineMatcher.group(1);
+            String by = deadlineMatcher.group(3);
+
+            LocalDateTime byDateTime = formatDateTime(by);
+
+            newTask = new Deadlines(task, byDateTime);
+        } else if (eventMatcher.matches()) {
+            String task = eventMatcher.group(1);
+            String start = eventMatcher.group(3);
+            String end = eventMatcher.group(5);
+
+            LocalDateTime startDateTime = formatDateTime(start);
+            LocalDateTime endDateTime = formatDateTime(end);
+
+            newTask = new Events(task, startDateTime, endDateTime);
+        } else if (toDoMatcher.matches()) {
+            String task = toDoMatcher.group(1);
+            newTask = new ToDos(task);
+        } else {
+            if (input.startsWith("deadline")) {
+                System.out.println(Constants.errorPrint.deadline());
+            } else if (input.startsWith("todo")) {
+                System.out.println(Constants.errorPrint.todo());
+            } else if (input.startsWith("event")) {
+                System.out.println(Constants.errorPrint.event());
+            } else if (input.isBlank()) {
+                System.out.println(Constants.errorPrint.erroneousAdd());
+            } else {
+                System.out.println(Constants.errorPrint.general());
+            }
+            return;
+        }
+
+        taskLog.add(newTask);
+        String addPrint = "-----------------------------------------\n"
+                + "    added to log: " + newTask + "\n"
+                + "    key in 'log' to view your current task log ^.^\n"
+                + "-----------------------------------------\n";
+        System.out.println(addPrint);
+        return;
+    }
+
+    static void viewLog() {
+        StringBuilder logPrint = new StringBuilder();
+        if (taskLog.isEmpty()) {
+            System.out.println(Constants.emptyLogPrint);
```

Comment:
Consider renaming this function to "printEmptyLog" to start the name with the verb